lib/auth/trustedcluster.go

```diff
@@ -46,28 +47,48 @@ func (a *AuthServer) UpsertTrustedCluster(trustedCluster services.TrustedCluster
 }
 
 func (a *AuthServer) enableTrustedCluster(trustedCluster services.TrustedCluster) error {
-	// get the certificate authorities for this auth server
-	localCertAuthorities, err := a.GetCertAuthorities(services.HostCA, false)
+	var localCertAuthorities []services.CertAuthority
+
+	// get a list of certificate authorities for this auth server
+	allLocalCAs, err := a.GetCertAuthorities(services.HostCA, false)
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	for _, lca := range allLocalCAs {
+		if lca.GetClusterName() == a.DomainName {
+			localCertAuthorities = append(localCertAuthorities, lca)
+		}
+	}
 
 	// create a request to validate a trusted cluster (token and local certificate authorities)
 	validateRequest := ValidateTrustedClusterRequest{
 		Token: trustedCluster.GetToken(),
 		CAs:   localCertAuthorities,
 	}
 
+	// log the local certificate authorities that we are sending
```

Comment:
seems like a lot of repetition of the same logic, can you factor this into separate function, e.g.

```go
type CertAuthorities []CertAuthority

func (s CertAuthorities) String() string {
  ... formatting logic
}

so later you can do this:

log.Debugf("[TRUSTED CLUSTER %v", CertAuthority(validateRequest.CAs))
```